src/libstd/io/impls.rs

```diff
@@ -76,6 +112,15 @@ impl<'a> Write for &'a mut [u8] {
         *self = b;
         Ok(amt)
     }
+
+    fn write_all(&mut self, data: &[u8]) -> io::Result<()> {
+        if try!(self.write(data)) == data.len() {
+            Ok(())
+        } else {
+            Err(Error::new(ErrorKind::WriteZero, "failed to write whole buffer", None))
```

Comment:
@sfackler: yeah, I agree. I kept the `WriteZero` because that's the error the `Read::write_all` supermethod returns. The fact that I'm not doing the loop is more of an implementation detail. Is it worth making a `PartialWrite` error kind?